### PR TITLE
HCCDOC-2724 | Document endpoints

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
## What?
I added the documentation that we have prepared along @emmurphy1.

[[HCCDOC-2724]](https://issues.redhat.com/browse/HCCDOC-2724)

## Why?
There is an ongoing effort to document our endpoints.

## How?
The changes were applied in the JSON file and then that file was converted to YAML, since the former seemed to be the most up to date one.

## Testing
N/A

## Anything Else?
N/A

### CODE REVIEW

**Review by gmcculloug — APPROVED** [review]
Looks good to me.  @dscervantes Please review.

**Comment by MikelAlejoBR** [comment]
@gmcculloug since it's documentation I'm going to go ahead and just merge it, especially because we've already reviewed the texts two or three times. We can always come back and fix the docs if we find any erratas or typos!

**Comment by MikelAlejoBR** [comment]
Well I am actually not going to merge it because I don't have permissions :joy:

File: static/spec/openapi.json [hunk 1]
```diff
@@ -549,6 +551,19 @@
...
+        "type": "object",
+        "properties": {
+          "message": {
+            "type": "string",
+            "example": "Something went wrong"
+          },
+          "code": {
+            "type": "string",
+            "example": "123"
+          }
```
Comment (dehort) [inline static/spec/openapi.json:565]: The code type should be int.  https://github.com/RedHatInsights/export-service-go/blob/main/exports/jsonerror_response.go#L18

File: static/spec/openapi.yaml [hunk 2]
```diff
@@ -86,10 +102,18 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ExportList'
+        '500':
+          description: Failed to get object
```
Comment (dehort) [inline static/spec/openapi.yaml:106]: I think we should make the http status code 500 descriptions more generic. Maybe something like "Unexpected server side error".

File: static/spec/openapi.yaml [hunk 3]
```diff
@@ -136,9 +188,14 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ExportStatus'
+        '500':
+          description: Error while encoding
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
```
Comment (dehort) [inline static/spec/openapi.yaml:196]: I think this one's description should be a bit more generic.

File: static/spec/openapi.yaml [hunk 4]
```diff
@@ -294,18 +351,24 @@ components:
...
               format: integer
+    ErrorResponse:
+      type: object
+      properties:
+        message:
+          type: string
+          example: Something went wrong
+        code:
+          type: string
+          example: '123'
```
Comment (dehort) [inline static/spec/openapi.yaml:362]: This type should be int.

**Review by dehort — CHANGES_REQUESTED** [review]
Looking good.  I left a few comments.

Thank you for the PR @MikelAlejoBR !

File: static/spec/openapi.json (hunk 1 above)
Comment (MikelAlejoBR) [reply static/spec/openapi.json]: Changed!

File: static/spec/openapi.yaml (hunk 2 above)
Comment (MikelAlejoBR) [reply static/spec/openapi.yaml]: Good idea, yes. Refactored!

File: static/spec/openapi.yaml (hunk 4 above)
Comment (MikelAlejoBR) [reply static/spec/openapi.yaml]: Changed!

File: static/spec/openapi.yaml (hunk 3 above)
Comment (MikelAlejoBR) [reply static/spec/openapi.yaml]: Changed!

**Pull request merged**